### ISSUE TITLE
Allow reading Auth0 client secret dynamically

### ISSUE THREAD

**Issue by alexzelenuyk** [auth0/nextjs-auth0#1490]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

I want to rotate the client secret of my Auth0 without downtime regularly.

Currently, the application reads `clientSecret` from the environment variable during the deployment. Thus, when the `clientSecrets` get rotated, there is no way to re-initialize Auth0 SDK without a new deployment. From the moment the secret is turned until the deployment, the application partially does not work, because the application "knows" only the old secret.

### Describe the ideal solution

I want to be able to change the `clientSecret` during the runtime. Ideally, for example, being able to implement reading the `clientSecret` from AWS Secrets Manager.

### Alternatives and current workarounds

I didn't find any feasible workarounds

### Additional context

There are several discussions in the Forum related to this problem: 

1. https://community.auth0.com/t/nextjs-auth0-async-credentials-and-userprovider/117395
2. https://community.auth0.com/t/how-to-use-aws-secrets-manager-with-auth0-nextjs-auth0/91838/3

**Comment by adamjmcgrath** [comment]
Hi @alexzelenuyk 

> Currently, the application reads clientSecret from the environment variable during the deployment. 

Have a look at https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md#create-your-own-instance-of-the-sdk

Creating your own instance of the SDK allows you to provide the `clientSecret` in code.

**Comment by alexzelenuyk** [comment]
Hey @adamjmcgrath ,

Thank you very much for your instance response!

This example works well for the first initialization. However, if `clientSecret` was rotated afterward, I would need to deploy my application again because `initAuth0()` could be called only once.

In this example, it would be nice to have `clientSecret`  parameter as a function, not as a static string. Thus, I can implement reading a secret from AWS Secrets manager. For example:

```
// utils/auth0.js
import { initAuth0 } from '@auth0/nextjs-auth0';

export default initAuth0({
  ...
  clientSecret: async () => {
   const clientSecret = await readFromAwsSM();
   return clientSecret
  }
});
```

This will enable a possibility to rotate secrets using, for example, AWS SM Secrets Rotation Lambda.

**Comment by adamjmcgrath** [comment]
Hi @alexzelenuyk - we can't provide that feature, but you could recreate the instance if the secret changes

```js
// utils/auth0.js
import { initAuth0 } from '@auth0/nextjs-auth0';

let auth0;

export default (req, res) => {
  if (!auth0 || hasNewClientSecret) {
    auth0 = initAuth0({ latestClientSecret })
  } 
  return auth0(req, res);
};
```

**Comment by alexzelenuyk** [comment]
Thank you very much, this looks like what I need.

In this case, I need to adjust `pages/api/auth/[auth0].js` from the example:

```
import auth0 from '../../../utils/auth0';

// add this line
const auth0instance = await auth0();

export default auth0instance.handleAuth();
```

Thank you very much for your help!

**Comment by alexzelenuyk** [comment]
Unfortunately I was too optimistic, and after another look, it didn't work: when you run the application in Docker, `pages/api/auth/[auth0].js` function will be called only once. Thus, when a secret is changed in the AWS Secret Manger, it won't be updated. 

Is there a way to reload an `auth0` config on every request when running in Docker?

**Comment by adamjmcgrath** [comment]
Hi @alexzelenuyk

> when you run the application in Docker, pages/api/auth/[auth0].js function will be called only once

have a look at my example https://github.com/auth0/nextjs-auth0/issues/1490#issuecomment-1761359169 

The function in that example will be called once for every request
